flexmeasures/ui/templates/_macros.html

```diff
@@ -278,12 +205,62 @@
     }
   ]
 }
+vegaEmbed("#view", treeSpecs, { renderer: "svg" }).then(function(result) {
+  result.view.addSignalListener("node_click", function(name, value) {
+    if (value) {
+      openModal(value);
+    }
+  });
+});
 
-    vegaEmbed(
-      "#view",
-      treeSpecs,
-      {renderer: "svg"}
-    );
+function openModal(data) {
```

Comment:
Awesome direction you're taking this in! It looks very clean to have the sensor listing in the modal..

![image](https://github.com/user-attachments/assets/b4788623-1e28-43b3-93bb-b4f790f9f1c7)